src/importlinter/domain/helpers.py

```diff
@@ -34,6 +34,26 @@ def pop_imports(
     return removed_imports
 
 
+def pop_import_expressions(
+    graph: ImportGraph, expressions: Iterable[ImportExpression]
+) -> List[Dict[str, Union[str, int]]]:
+    imports: List[DirectImport] = []
+    for expression in expressions:
```

Comment:
Nice. I think it would be worth extracting lines 41 - 53 into a separate `import_expressions_to_imports` function, which would be a side-effect free function so easier to test.

It would be great to have unit test coverage too. If I were you I would focus the unit testing on the pure function `import_expressions_to_imports` - possibly don't need one for `pop_import_expressions` if we have coverage on the contract types instead.